docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+### sigstore config
+
+The configuration file that specifies the container image sigstore is specified as all `.yaml` files in /etc/containers/registries.d directory. After the policy.json allowlist check is completed, the entries in the sigstore configuration file will claim where the container image is located, and the signature claim file of the container image will be downloaded from the specified URL (a webserver or a local directory). The sample sigstore configuration file matching the sample policy.json file in the above is as follows:
+
+```yaml
+default:
+    sigstore: file:///var/lib/containers/sigstore
+docker:
+  docker.io/my_private_registry:
+    sigstore: https://my-sigstore.com/example/sigstore
+  registry.access.redhat.com:
+    sigstore: https://access.redhat.com/webassets/docker/content/sigstore
+```
+
+If there is no `.yaml` file in the /etc/containers/registries.d/ directory when image-rs want to read it, as the way to obtain policy.json, image-rs should call the GetResource grpc service provided by the attestation-agent. After remote attestation, it is obtained from the user controlled relying KBS through the encrypted channel.
+
+### Image signature verification
+
+After completing the white list check and signature file download, signature verification is divided into the following steps:
+
+1. Parse the signature claim file and separate the JSON payload and cryptographic signature.
+
+2. Read the policy.json's "signedBy" policy, get the public key used to verify the signature from the "keyPath" or "keyData" field.
+
+3. Use the public key to verify the cryptographic signature in the signature claim file, and then compare whether the manifest digest in the JSON payload is consistent with that of the container image and whether the docker reference in the JSON payload is consistent with that of the container image.
+
+### Image decryption
+
+After verifying the container image signature, will start the actual image pulling. When pulling the image layer by layer, if encounter an encrypted image layer, image-rs calls the container image cryptography operation library provided by ocicrypt-rs to decrypt the encrypted image layer as follows:
+
+1. Read the layer's two annotations fields in the image manifest. The `org.opencontainers.image.enc.keys.provider.attestation` field is the annotation packet mentioned above, which contains `PrivateLayerBlockCipherOptions` encrypted by the user's private key. The symmetric key used to encrypt this layer is recorded in this `PrivateLayerBlockCipherOptions`.
+2. ocicrypt-rs calls the keyprovider grpc service provided by the attestation-agent and sends the annotation packet to the attestation-agent.
+3. The attestation-agent performs HW-TEE remote attestation with the relying KBS deployed by the remote user. After passing the attestation, it interacts with KBS through the encrypted channel, and decrypts the `PrivateLayerBlockCipherOptions` using the user's public key in the KBS.
```

Comment:
First use of HW-TEE acronym, expand and add link like for KBS above.

Reply:
"after passing the attestation" is a bit ambiguous, since "pass" can mean both "passing some test successfully" or "passing some information". I would just write "after attestation".

Reply:
+1. Pass can also mean to skip or ignore, so it's arguably a (semi-)auto-antonym.